packages/serverless-contracts/src/contracts/apiGateway/features/mockHandlerInput.ts

```diff
@@ -0,0 +1,99 @@
+import merge from 'lodash/merge';
+
+import {
+  getAPIGatewayEventRequestContextMock,
+  getAPIGatewayV2EventRequestContextMock,
+} from '@swarmion/serverless-helpers';
+
+import { GenericApiGatewayContract } from '../apiGatewayContract';
+import {
+  ApiGatewayEvent,
+  BodyType,
+  CustomRequestContextType,
+  HandlerCallback,
+  HandlerEventType,
+  HeadersType,
+  PathParametersType,
+  QueryStringParametersType,
+} from '../types';
+import {
+  ApiGatewayAuthorizerType,
+  ApiGatewayIntegrationType,
+} from '../types/constants';
+import { getHandlerContextMock } from '../utils/mockHandlerContext';
+
+export const getMockHandlerInput = <
+  Contract extends GenericApiGatewayContract,
+  IntegrationType extends ApiGatewayIntegrationType = Contract['integrationType'],
+  AuthorizerType extends ApiGatewayAuthorizerType = Contract['authorizerType'],
+  AdditionalArgs extends unknown[] = never[],
+  CustomEvent = HandlerEventType<
+    IntegrationType,
+    AuthorizerType,
+    PathParametersType<Contract>,
+    QueryStringParametersType<Contract>,
+    HeadersType<Contract>,
+    CustomRequestContextType<Contract>,
+    BodyType<Contract>
+  >,
+  Event = ApiGatewayEvent<IntegrationType, AuthorizerType>,
+  Callback = HandlerCallback<IntegrationType>,
+>(
+  contract: Contract,
+  partialEvent: Partial<CustomEvent> = {},
+  ...additionalArgs: AdditionalArgs
+): [Event, Context, Callback, ...AdditionalArgs] => {
+  const defaultBody =
+    contract.bodySchema !== undefined
+      ? JSONSchemaFaker.generate(contract.bodySchema as Schema)
+      : undefined;
+
+  const defaultPathParameters =
+    contract.pathParametersSchema !== undefined
+      ? JSONSchemaFaker.generate(contract.pathParametersSchema as Schema)
+      : undefined;
+
+  const defaultQueryStringParameters =
+    contract.queryStringParametersSchema !== undefined
+      ? JSONSchemaFaker.generate(contract.queryStringParametersSchema as Schema)
+      : undefined;
+
+  const defaultHeaders =
+    contract.headersSchema !== undefined
+      ? JSONSchemaFaker.generate(contract.headersSchema as Schema)
+      : undefined;
+
+  const defaultRequestContext =
+    contract.integrationType === 'restApi'
+      ? getAPIGatewayEventRequestContextMock()
+      : getAPIGatewayV2EventRequestContextMock();
+
+  const defaultCustomRequestContext =
+    contract.requestContextSchema !== undefined
+      ? JSONSchemaFaker.generate(contract.requestContextSchema as Schema)
+      : {};
+
+  const defaultEvent = {
+    pathParameters: defaultPathParameters,
+    queryStringParameters: defaultQueryStringParameters,
+    headers: defaultHeaders,
+    body: defaultBody,
+    requestContext: merge(defaultRequestContext, defaultCustomRequestContext),
+  };
+
+  const event = merge(defaultEvent, partialEvent);
+
+  const stringifiedEvent = {
+    ...event,
+    body: event.body !== undefined ? JSON.stringify(event.body) : undefined,
+  };
+
+  return [
+    stringifiedEvent as Event,
```

Comment:
naive question ? Do you need the cast here ? Isn't explicitely typing `stringifiedEvent` as `Event` not enough ?

Reply:
'Event' could be instantiated with an arbitrary type which could be unrelated to
:'(